### PR TITLE
DOC - add example to RegularGridInterpolator

### PR DESCRIPTION
**Pull request by sbyrnes321**
I just posted an example of RegularGridInterpolator [on stackoverflow this morning](http://stackoverflow.com/questions/21836067/interpolate-3d-volume-with-numpy-and-or-scipy/28858633#28858633), and then it occurred to me that it would be nice to have a similar example usage in the docs.

(Before accepting this PR, someone should please check that it builds properly with the doc-generator and that it passes doctests if applicable. Sorry that I am not doing that myself ... it seems complicated to set up. :-P)

### CODE REVIEW

File: scipy/interpolate/interpolate.py [hunk 1]
```diff
@@ -1464,6 +1464,36 @@ class RegularGridInterpolator(object):
...
+
+    >>> from scipy.interpolate import RegularGridInterpolator
+    >>> def f(x,y,z):
+    >>>     return 2 * x**3 + 3 * y**2 - z
+    >>> x = np.linspace(1, 4, 11)
+    >>> y = np.linspace(4, 7, 22)
+    >>> z = np.linspace(7, 9, 33)
+    >>> data = np.zeros((11, 22, 33))
+    >>> for i in range(11):
+    >>>     for j in range(22):
```
Comment (ewmoore) [inline scipy/interpolate/interpolate.py]: Let's use broadcasting here instead.  
```
data = f(x[:, np.newaxis, np.newaxis], y[np.newaxis, :, np.newaxis], z[np.newaxis, np.newaxis,:])
```
or equivalent.

Comment (pv) [inline scipy/interpolate/interpolate.py]: meshgrid(x, y, z, indexing='ij', sparse=True)

Comment (ewmoore) [inline scipy/interpolate/interpolate.py]: What he said.

**Comment by sbyrnes321** [comment]
OK, I have now gotten rid of the for loops.

File: scipy/interpolate/interpolate.py [hunk 2]
```diff
@@ -1464,6 +1464,33 @@ class RegularGridInterpolator(object):
 
     .. versionadded:: 0.14
 
+    Example
```
Comment (rgommers) [inline scipy/interpolate/interpolate.py]: This should be ``Examples``. That's the section header name recognized by the doc builder. (and add the corresponding underscore on the next line)

File: scipy/interpolate/interpolate.py [hunk 3]
```diff
@@ -1464,6 +1464,33 @@ class RegularGridInterpolator(object):
...
+
+    Evaluate a simple example function on the points of a 3D grid:
+
+    >>> from scipy.interpolate import RegularGridInterpolator
+    >>> def f(x,y,z):
+    >>>     return 2 * x**3 + 3 * y**2 - z
+    >>> x = np.linspace(1, 4, 11)
+    >>> y = np.linspace(4, 7, 22)
+    >>> z = np.linspace(7, 9, 33)
+    >>> # Our "data" array is defined by data[i,j,k] = f(x[i], y[j], z[k]):
```
Comment (rgommers) [inline scipy/interpolate/interpolate.py]: Preferably write this as a normal sentence, not as a comment in the interpreter. The code sample should be surrounded by double backticks and there should be blank lines above and below the sentence.

File: scipy/interpolate/interpolate.py [hunk 4]
```diff
@@ -1464,6 +1464,33 @@ class RegularGridInterpolator(object):
...
+    >>> data = f(*np.meshgrid(x,y,z,indexing='ij',sparse=True))
+
+    Define an interpolating function from this data:
+
+    >>> my_interpolating_function = RegularGridInterpolator((x,y,z), data)
+
+    Evaluate the interpolating function at the two points
+    (x,y,z) = (2.1, 6.2, 8.3) and (3.3, 5.2, 7.1):
```
Comment (rgommers) [inline scipy/interpolate/interpolate.py]: same here, double backticks for code samples

File: scipy/interpolate/interpolate.py [hunk 5]
```diff
@@ -1464,6 +1464,33 @@ class RegularGridInterpolator(object):
...
+
+    >>> pts = np.array([[2.1, 6.2, 8.3], [3.3, 5.2, 7.1]])
+    >>> my_interpolating_function(pts)
+    array([ 125.80469388,  146.30069388])
+    
+    which is indeed a close approximation to [f(2.1,6.2,8.3), f(3.3,5.2,7.1)].
```
Comment (rgommers) [inline scipy/interpolate/interpolate.py]: and here double backticks as well

File: scipy/interpolate/interpolate.py [hunk 6]
```diff
@@ -1464,6 +1464,33 @@ class RegularGridInterpolator(object):
...
+    Evaluate a simple example function on the points of a 3D grid:
+
+    >>> from scipy.interpolate import RegularGridInterpolator
+    >>> def f(x,y,z):
+    >>>     return 2 * x**3 + 3 * y**2 - z
+    >>> x = np.linspace(1, 4, 11)
+    >>> y = np.linspace(4, 7, 22)
+    >>> z = np.linspace(7, 9, 33)
+    >>> # Our "data" array is defined by data[i,j,k] = f(x[i], y[j], z[k]):
+    >>> data = f(*np.meshgrid(x,y,z,indexing='ij',sparse=True))
```
Comment (rgommers) [inline scipy/interpolate/interpolate.py]: PEP8: spaces after commas

**Comment by rgommers** [comment]
It will build with the minor changes I commented on. We don't run doctests for scipy, so no worries about those.

**Comment by sbyrnes321** [comment]
Thanks! I just added another commit trying to follow all your good suggestions.

**Comment by rgommers** [comment]
Squashed and merged in commit 25e741e. Thanks @sbyrnes321!

**Pull request closed without merging**